ufo2fontir/src/source.rs

```diff
@@ -519,6 +519,48 @@ fn glyph_order(
     Ok(glyph_order)
 }
 
+fn postscript_names(lib_plist: &plist::Dictionary) -> Result<PostscriptNames, WorkError> {
+    let postscript_names = match lib_plist.get("public.postscriptNames") {
+        Some(value) => {
+            let postscript_names_lib = value.as_dictionary().ok_or_else(|| {
+                WorkError::ParseError(
+                    PathBuf::from("lib.plist"),
+                    String::from("public.postscriptNames isn't a dictionary"),
+                )
+            })?;
+            postscript_names_lib
+                .iter()
+                .filter_map(|(glyph_name, ps_name)| match ps_name.as_string() {
+                    Some(ps_name) => Some((
+                        GlyphName::from(glyph_name.as_str()),
+                        GlyphName::from(ps_name),
+                    )),
+                    None => {
+                        warn!("public.postscriptNames: \"{glyph_name}\" has a non-string entry");
+                        None
+                    }
+                })
+                .collect()
+        }
+        None => HashMap::new(),
+    };
+    // Warn about duplicate public.postscriptNames values
+    // Allocate space ahead of time for speed, and because in the happy path the set length will be
+    // the same as the map len
+    let mut duplicate_values_check = HashSet::with_capacity(postscript_names.len());
+    let duplicate_values = postscript_names
+        .values()
+        .filter(|ps_name| {
+            let ps_name = (*ps_name).clone();
+            !duplicate_values_check.insert(ps_name)
+        })
+        .collect::<BTreeSet<_>>();
+    if !duplicate_values.is_empty() {
+        warn!("public.postscriptNames: the following production names are used by multiple glyphs: {duplicate_values:?}");
```

Comment:
Since we don't use duplicate_values at all if warn is suppressed I would wrap the dup checking in `if log::log_enabled!(log::Level::Warn) {`